src/driver/better-sqlite3/BetterSqlite3Driver.ts

```diff
@@ -79,6 +80,33 @@ export class BetterSqlite3Driver extends AbstractSqliteDriver {
         return super.normalizeType(column);
     }
 
+    async afterConnect(): Promise<void> {
+        return this.attachDatabases();
+    }
+
+    /**
+     * For SQLite, the database may be added in the decorator metadata. It will be a filepath to a database file.
+     */
+    buildTableName(tableName: string, _schema?: string, database?: string): string {
+
+        if (!database) return tableName;
+        if (this.attachedDatabases[database]) return `${this.attachedDatabases[database].attachHandle}.${tableName}`;
+
+        if (database === this.options.database) return tableName;
+
+        // we use the decorated name as supplied when deriving attach handle (ideally without non-portable absolute path)
+        const identifierHash = filepathToName(database);
+        // decorated name will be assumed relative to main database file when non absolute. Paths supplied as absolute won't be portable
+        const absFilepath = isAbsolute(database) ? database : path.join(this.getMainDatabasePath(), database);
+
+        this.attachedDatabases[database] = {
```

Comment:
this seems to be a side effect of `buildTableName`. Can't we do this "population" logic in a different "initialization phase" place?

Reply:
(so here we can simply return the name based on this.attachedDatabases if needed)

Reply:
Do you have any suggestions which code could trigger this initialization?

For it to work, I need to ensure that all tables will be passed over (Including migration cli etc also)